### DISCUSSION TITLE
useFetch null

### DISCUSSION THREAD

**Discussion by Sandr0oo** [nuxt/nuxt#18892] [Questions]
Hello. Help me please.
I don't understand why HotPricechanges.Data = Null? I use Server: False.
![Code_PV2u3f0mY0](https://user-images.githubusercontent.com/61181414/217537543-26055314-e3e0-467d-b07f-836c82868529.png)

![Code_SsU6VckEQ3](https://user-images.githubusercontent.com/61181414/217537607-05d1461d-8dbc-4ec0-b150-80e0977b0bdf.png)

**Comment by pi0** [comment]
Check `HotPricechanges.error` a server error might have happened.

**Reply by Sandr0oo** [reply]
If I add a check for NULL in the cycle, then such an error does not occur. But if you don't do the check, there will be an error.
![изображение](https://user-images.githubusercontent.com/61181414/217557089-4e8a1c44-ed70-4633-a742-69fdf0e0ffec.png)

**Comment by AloisSeckar** [comment]
the error object doesn't reside inside `data`...you need to deconstruct `useFetch` result like that:

```js
const { data: hotPriceChanges, error } = await getHotChanges();
console.log(error)
```

**Reply by Sandr0oo** [reply]
![изображение](https://user-images.githubusercontent.com/61181414/217630485-5660341e-8451-45a5-a521-454a96046fa4.png)

**Reply by AloisSeckar** [reply]
hm...not 100% sure, but I think you also need to return the actuall error from your onRequestError and onResponseError? I haven't worked with fetch interceptors yet, but looking at the code, I'd say, you are swallowing it inside those interceptors

**Reply by Sandr0oo** [reply]
I have error output to the console in onRequestError and onResponseError. But as you can see in the screenshot above, when you call useFetch the events onRequestError and onResponseError do not happen.
![изображение](https://user-images.githubusercontent.com/61181414/217817396-3a068d08-e079-4ae4-914e-596eb84da33d.png)

**Reply by AloisSeckar** [reply]
I would say this discussion #13805 might be related to your situation. Because of the `server: false` option the fetch ain't actually blocking the async-await code and since you then immideately try to read the data, it crashes.

I would try to switch `function chunkedPriceChanges()` to `const chunkedPriceChanges = computed(() =>` and add null-checks to ensure it wont initially fail: `this.hotPriceChanges?.data?.priceChanges?.length`

**Reply by Sandr0oo** [reply]
Thank you so much!
